sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample07_RestartProcessongOnError.cs

```diff
@@ -0,0 +1,231 @@
+                // Processor client is able to perform its operations efficiently.
+                //
+                // The process error handler is not awaited by the Event Processor client and is, instead, executed in a fire-and-forget manner.  This
+                // means that you may safely interact with the Event Processor client, such as requesting that it stop processing.
+
+                if (eventArgs.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                // Because there is no long-running or I/O operation, inspecting the cancellation token again does not make sense in this scenario.
+                // However, in real-world processing, it is recommended that you do so as you are able without compromising your ability to capture
+                // and troubleshooting information.
+                //
+                // It is also recommended that the cancellation token be passed to any asynchronous operations that are awaited in this handler.
+
+                Console.WriteLine();
+                Console.WriteLine("===============================");
+                Console.WriteLine($"The error handler was invoked during the operation: { eventArgs.Operation ?? "Unknown" }, for Exception: { eventArgs.Exception.Message }");
+                Console.WriteLine("===============================");
+                Console.WriteLine();
+
+                // We will not pass the cancellation token from the event arguments here, as it will be
+                // signaled when we request that the processor stop.
+
+                if (eventArgs.Exception is EventHubsException)
+                {
+                    Console.WriteLine("Detected an service error.  Restarting the processor...");
+
+                    await processor.StopProcessingAsync();
+                    await processor.StartProcessingAsync();
+
+                    Console.WriteLine("Processor has been restarted....");
+                    Console.WriteLine();
+                }
+            }
+
+            processor.ProcessEventAsync += processEventHandler;
+            processor.ProcessErrorAsync += processErrorHandler;
+
+            try
+            {
+                Console.WriteLine("Starting the Event Processor client...");
+
+                eventIndex = 0;
+                await processor.StartProcessingAsync();
+
+                using var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+                // Unfortunately, because the handler is invoked when exceptions are encountered in the Event Processor client and not for the code
+                // developers write in the event handlers, there is no reliable way to force an exception.  As a result, it is unlikely that you will
+                // be able to observe the error handler in action by just running the Event Processor.
```

Comment:
My fear is that if we induce an exception by creating another consumer to compete, then we're opening the door to confusion for those that haven't worked with the core library and don't have a conceptual understanding of owner levels and exclusive vs. non-exclusive consumers.  You're right in that it would, at least, be a natural error trigger, but the complexity is concerning.

The connection string is an interesting idea.   That hadn't occurred to me, but I'm not sure how we would be able to successfully restart the processor if we fed it an invalid connection string; we'd just fail forever in a loop.   

I'm definitely open to ideas here; I don't like the artificial trigger, but without dropping to test-injection tricks that wouldn't be valid in a real world case and would come with their own orthogonal complexity cost.... I don't have a good idea.

I'm going to leave as-is for now, but if you have some ideas, please ping me and we can potentially do a follow-up revision.